### PR TITLE
test(contrib/drivers/mysql): add pagination and error handling tests

### PR DESCRIPTION
**Pull request by lingcoder**
## Summary
- Add comprehensive pagination tests (Limit, Offset, Page, ForPage)
- Add error handling tests for invalid operations
- Add tests for edge cases and boundary conditions

**Test coverage added:**
- Pagination: ~28 test functions
- Error handling: ~20 test functions

Ref #4689

## Test plan
```bash
cd contrib/drivers/mysql
go test -v -run "TestModel_Pagination|TestModel_Error|TestModel_InvalidOperation"
```

### CODE REVIEW

File: contrib/drivers/mysql/mysql_z_unit_feature_error_handling_test.go [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+
+// Test_Model_Value_EmptyResult tests Value with empty result
+func Test_Model_Value_EmptyResult(t *testing.T) {
+	table := createInitTable()
+	defer dropTable(table)
+
+	gtest.C(t, func(t *gtest.T) {
+		value, err := db.Model(table).Where("id > ?", 1000).Value()
+		t.AssertNil(err)
+		t.Assert(value.IsEmpty(), true)
```
[the remark this answers is not in the document]
Comment (lingcoder) [reply contrib/drivers/mysql/mysql_z_unit_feature_error_handling_test.go:335]: 已验证：`gvar.Var.Val()` 方法在 `gvar_basic.go:16` 有 nil receiver 检查（`if v == nil { return nil }`），因此 `value.IsEmpty()` 对 nil `*gvar.Var` 是安全的，不会 panic。无需修改。
